src/olympia/versions/models.py

```diff
@@ -196,16 +196,18 @@ def should_have_due_date(self, negate=False):
         )
         is_pre_review_version = Q(
             Q(file__status=amo.STATUS_AWAITING_REVIEW)
+            & ~Q(addon__status=amo.STATUS_DELETED)
```

Comment:
this _shouldn't_ be necessary as deleting an add-on should set the file statuses to disabled but it's clearer that the `is_pre_review_version` `Q` is restricted to non-deleted add-ons.